test/parallel/test-assert-deep.js

```diff
@@ -890,3 +890,10 @@ assert.deepStrictEqual(obj1, obj2);
   );
   util.inspect.defaultOptions = tmp;
 }
+
+// Basic array out of bounds check.
```

Comment:
Add test for the negative index use case?

```js
{
  const arr = Array(2);
  arr[0] = 1;
  arr[-1] = 1;
  assertNotDeepOrStrict(arr, [1, undefined]);
}
```

Or something similar

Reply:
This is already covered.